lib/code_sponsor_web/controllers/ad_serve_controller.ex

```diff
@@ -0,0 +1,73 @@
+defmodule CodeSponsorWeb.AdServeController do
+  use CodeSponsorWeb, :controller
+
+  alias CodeSponsor.{Properties, Sponsorships, Creatives}
+
+  def embed(conn, %{"property_id" => property_id} = params) do
+    property      = Properties.get_property!(property_id)
+    template_slug = params["template"] || "default"
+    theme_slug    = params["theme"] || "light"
+    template      = Creatives.get_template_by_slug(template_slug)
+    targetId      = params["target"] || "codefund_ad"
+
+    cond do
```

Comment:
I think this could be easily refactored around a `with` clause, try building it off your success case and use the `else`s to match on your `nil` cases.

Reply:
I'm afraid my skills are not at this point yet :(